File: shoring/blackjackEngine/src/test/java/com/panopset/tests/blackjackEngine/SplitNotAllowedTest.java
package com.panopset.tests.blackjackEngine;

import static com.panopset.blackjackEngine.CommandDefinitions.CMD_DEAL;
import static com.panopset.blackjackEngine.CommandDefinitions.CMD_RESET;
import static com.panopset.blackjackEngine.CommandDefinitions.CMD_SPLIT;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import com.panopset.blackjackEngine.BlackjackConfigDefault;
import com.panopset.blackjackEngine.BlackjackGameEngine;
import com.panopset.blackjackEngine.CardDefinition;

public class SplitNotAllowedTest extends AbstractBlackjackTest {

  @Test
  void test() {
    BlackjackGameEngine bge = new BlackjackGameEngine(new BlackjackConfigDefault());
    bge.exec(CMD_RESET);
    bge.getShoe().stackTheDeckFromList(splitNotAllowedTest());
    bge.exec(CMD_DEAL);
    bge.exec(CMD_SPLIT);
    
    verifyPlayerCards(new CardDefinition[] {CardDefinition.TWO_OF_SPADES, CardDefinition.EIGHT_OF_DIAMONDS}, 
        bge.getCurrentSnapshot().getPlayers().get(0).getHands().get(0).getCards());
    
    Assertions.assertEquals(CardDefinition.TWO_OF_SPADES.getFace(), bge.getCurrentSnapshot().getPlayers().get(0).getActiveHand().getCards().get(0).getCard().getFace());
    Assertions.assertEquals(CardDefinition.TWO_OF_SPADES.getSuit(), bge.getCurrentSnapshot().getPlayers().get(0).getActiveHand().getCards().get(0).getCard().getSuit());
    Assertions.assertEquals(CardDefinition.EIGHT_OF_DIAMONDS.getFace(), bge.getCurrentSnapshot().getPlayers().get(0).getActiveHand().getCards().get(1).getCard().getFace());
    Assertions.assertEquals(CardDefinition.EIGHT_OF_DIAMONDS.getSuit(), bge.getCurrentSnapshot().getPlayers().get(0).getActiveHand().getCards().get(1).getCard().getSuit());
    Assertions.assertEquals("Can't split cards that don't have the same face",
        bge.getDealerMessage());
  }

}

File: code/shoring/blackjackEngine/src/test/kotlin/com/panopset/tests/blackjackEngine/SplitNotAllowedTest.kt
package com.panopset.tests.blackjackEngine

import com.panopset.blackjackEngine.BlackjackConfigDefault
import com.panopset.blackjackEngine.BlackjackGameEngine
import com.panopset.blackjackEngine.CardDefinition
import com.panopset.blackjackEngine.*
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class SplitNotAllowedTest {
    @Test
    fun test() {
        val bge = BlackjackGameEngine(BlackjackConfigDefault())
        bge.exec(CMD_RESET)
        bge.getShoe().stackTheDeckFromList(splitNotAllowedTest())
        bge.exec(CMD_DEAL)
        bge.exec(CMD_SPLIT)
        verifyPlayerCards(
            arrayOf(CardDefinition.TWO_OF_SPADES, CardDefinition.EIGHT_OF_DIAMONDS),
            bge.getLatestSnapshot().players[0].hands[0].getBlackjackCards()
        )
        Assertions.assertEquals(
            CardDefinition.TWO_OF_SPADES.face,
            bge.getLatestSnapshot().players[0].activeHand!!.getBlackjackCards()[0].card.face
        )
        Assertions.assertEquals(
            CardDefinition.TWO_OF_SPADES.suit,
            bge.getLatestSnapshot().players[0].activeHand!!.getBlackjackCards()[0].card.suit
        )
        Assertions.assertEquals(
            CardDefinition.EIGHT_OF_DIAMONDS.face,
            bge.getLatestSnapshot().players[0].activeHand!!.getBlackjackCards()[1].card.face
        )
        Assertions.assertEquals(
            CardDefinition.EIGHT_OF_DIAMONDS.suit,
            bge.getLatestSnapshot().players[0].activeHand!!.getBlackjackCards()[1].card.suit
        )
        Assertions.assertEquals(
            "Can't split cards that don't have the same face",
            bge.dealerMessage
        )
    }
}
